hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/EnhancerImpl.java

```diff
@@ -539,4 +536,35 @@ else if ( access != null && access.loadSilent().value() == AccessType.FIELD ) {
 			}
 		}
 	}
+
+	private class EnhancerClassFileLocator extends ClassFileLocator.ForClassLoader {
+
+		private Map<String,Resolution> nameToResolutionMap = new HashMap<>();
```

Comment:
I suppose that instead of clearing the Map, we don't really need a map, but to instead just keep the name + bytes set during the ctor and create a new EnhancerClassFileLocator for each call to enhance().  The instead of nameToResultionMap.get(name), we could just see if saved name is equal to the name parameter, if yes, return new Resolution.Explicit( bytes).